src/libstd/sys/common/net.rs

```diff
@@ -221,11 +226,12 @@ impl TcpStream {
 
     pub fn write(&self, buf: &[u8]) -> io::Result<usize> {
         let len = cmp::min(buf.len(), <wrlen_t>::max_value() as usize) as wrlen_t;
+        let flags = if cfg!(target_os = "linux") { MSG_NOSIGNAL } else { 0 };
```

Comment:
Oh passing it there sometimes and not others is actually somewhat unrelated. Here yeah we should always pass `MSG_NOSIGNAL`